module5/TaskListApp/TaskListApp/Views/TaskDetailsView.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  TaskDetailsView.swift
+//  TaskListApp
+//
+//  Created by Peter Rice on 12/22/23.
+//
+
+import SwiftUI
+
+struct TaskDetailsView: View {
+  var tasksStore: TasksStore
+  let task: Task  
+  @State private var taskCompleted: Bool = false
+
+  var body: some View {
+    VStack {
+      Form {
+        Section(header:
+          Text("Task Title")
+        )
+        {
+          Text(task.title)
+            .bold()
+        }
+        Section(header: Text("Notes")) {
+          Text(task.notes)
+            .bold()
+        }
+        Section() {
+          Toggle(isOn: $taskCompleted) {
+            Text("Completed: ")
+              .bold()
+          }
+          .onChange(of: taskCompleted) {
+            tasksStore.setTask(with: task.id, completed: taskCompleted)
+          }
+        }
+      }
+    }    
+    .onAppear {
+      taskCompleted = task.isCompleted
```

Comment:
This causes your `.onChange` to fire every time we open this view with a completed task and it resets the task to be completed. consider using the new format of `onChange(of: taskCompleted) { oldValue newValue in }` and don't make the call if they are equal. Or better yet, set an initial value using `.onChange(of: taskCompleted, initial: task.isCompleted)`

Reply:
Hmm good suggestion.